### PR TITLE
EQ-376 Author Infrastructure

### PR DESCRIPTION
**Pull request by warrenbailey**
**_What**_
1. Create a VPC for the author application
2. Create Subnets for the application and database
3. Create the Elastic beanstalk application for the author to deploy into
4. Create a Postgres database for the application
5. Create security group to whitelist IP to the ONS range.

I've also restructured the Survey Runner terraform files so that they all exist inside a survey-runner folder

**_How to test**_
1. Check out this branch
2. cd into author
3. Update the terraform.tfvars with the correct values for your environment
4. run `terraform apply`
5. Check you can hit the example application in elastic beanstalk.

**_Who can review**_
Anyone apart from @warrenbailey

### CODE REVIEW

File: author/vpc.tf [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  vpc_id                  = "${aws_vpc.author-vpc.id}"
+  cidr_block              = "${var.application_cidr}"
+  availability_zone       = "eu-west-1a"
+  tags {
+    Name = "${var.env}-author-application-subnet"
+  }
+}
+
+# Create a subnet to launch our database into.
+resource "aws_subnet" "database-1" {
```
Comment (dhilton) [inline author/vpc.tf]: Might be helpful to rename these to `database-1c` and `database-1b` so we can easily grok what the subnets map to, as the only reason we have two is for the cross availability zone support.

**Comment by dhilton** [comment]
The one issue I can see with the IAM role support we're making use of here for S3 access, is that it's a bit of Amazon specific magic that we can't port to another PaaS provider easily; we'll need to rewrite the connection logic. It's not a massive issue and I think this approach is the correct one for AWS, I just think we need to be aware of that issue so we know the cost of changing provider. At first pass looks all good, just deploying now.

File: author/terraform.tfvars.example [hunk 2]
```diff
@@ -0,0 +1,11 @@
+env="xxxx"
```
Comment (dhilton) [inline author/terraform.tfvars.example]: I'm not sure we should set the `env` by default in the file as this needs to change dependent on user / environment as we don't want other developers clashing in our AWS accounts.

Comment (warrenbailey) [inline author/terraform.tfvars.example]: No worries I'll remove it :)

File: author/vpc.tf (hunk 1 above)
Comment (warrenbailey) [inline author/vpc.tf]: I'll change it now

File: author/terraform.tfvars.example [hunk 3]
```diff
@@ -0,0 +1,10 @@
+aws_access_key="xxxxxx"
+aws_secret_key="xxxxxx"
+aws_key_pair="xxxx"
+ons_access_ips="0.0.0.0/32"
```
Comment (dhilton) [inline author/terraform.tfvars.example]: Can we change this to `0.0.0.0/0` this caused my build to fail.

**Comment by dhilton** [comment]
👍 LGTM

**Pull request merged**